docker/dockerfile.merlin

```diff
@@ -169,8 +170,9 @@ ENV DEBIAN_FRONTEND=noninteractive
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/lib:/repos/dist/lib
 ENV PATH=${CUDA_HOME}/lib64/:${PATH}:${CUDA_HOME}/bin
 
+RUN sed -i -e 's/http:\/\/archive\.ubuntu\.com\/ubuntu\//mirror:\/\/mirrors\.ubuntu\.com\/mirrors\.txt/' /etc/apt/sources.list
 # Set up NVIDIA package repository
-RUN apt update -y --fix-missing && \
+RUN v=8.5.1-1+cuda11.8 && apt update -y --fix-missing && \
```

Comment:
This requires a --allow-downgrades flag during run. It fails otherwise. At least when I try to run the docker build command on this that happens.

Reply:
This is actually required for the last apt install command in this block

Reply:
Using 22.12 i get the following error: 
```In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:28,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp: In member function 'TRITONSERVER_Error* triton::backend::hps::HPSBackend::ParseParameterServer(const string&)':
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:169:40: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'max_get_batch_size'; did you mean 'max_batch_size'?
  169 |         TritonJsonHelper::parse(params.max_get_batch_size, json, key, false));
      |                                        ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/build/_deps/repo-backend-src/include/triton/backend/backend_common.h:97:38: note: in definition of macro 'RETURN_IF_ERROR'
   97 |     TRITONSERVER_Error* rie_err__ = (X); \
      |                                      ^
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:33,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:172:44: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'max_get_batch_size'; did you mean 'max_batch_size'?
  172 |         "max. batch size (GET) = ", params.max_get_batch_size);
      |                                            ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/include/triton_common.hpp:39:45: note: in definition of macro 'HPS_TRITON_LOG'
   39 |     const std::string& msg = hps_str_concat(__VA_ARGS__);               \
      |                                             ^~~~~~~~~~~
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:28,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:176:40: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'max_set_batch_size'; did you mean 'max_batch_size'?
  176 |         TritonJsonHelper::parse(params.max_set_batch_size, json, key, false));
      |                                        ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/build/_deps/repo-backend-src/include/triton/backend/backend_common.h:97:38: note: in definition of macro 'RETURN_IF_ERROR'
   97 |     TRITONSERVER_Error* rie_err__ = (X); \
      |                                      ^
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:33,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:179:44: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'max_set_batch_size'; did you mean 'max_batch_size'?
  179 |         "max. batch size (SET) = ", params.max_set_batch_size);
      |                                            ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/include/triton_common.hpp:39:45: note: in definition of macro 'HPS_TRITON_LOG'
   39 |     const std::string& msg = hps_str_concat(__VA_ARGS__);               \
      |                                             ^~~~~~~~~~~
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:28,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:184:16: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'refresh_time_after_fetch'
  184 |         params.refresh_time_after_fetch, json, key, false));
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/build/_deps/repo-backend-src/include/triton/backend/backend_common.h:97:38: note: in definition of macro 'RETURN_IF_ERROR'
   97 |     TRITONSERVER_Error* rie_err__ = (X); \
      |                                      ^
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:33,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:187:47: error: 'struct HugeCTR::VolatileDatabaseParams' has no member named 'refresh_time_after_fetch'
  187 |         "refresh time after fetch = ", params.refresh_time_after_fetch);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/include/triton_common.hpp:39:45: note: in definition of macro 'HPS_TRITON_LOG'
   39 |     const std::string& msg = hps_str_concat(__VA_ARGS__);               \
      |                                             ^~~~~~~~~~~
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:28,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:262:40: error: 'struct HugeCTR::PersistentDatabaseParams' has no member named 'max_get_batch_size'; did you mean 'max_batch_size'?
  262 |         TritonJsonHelper::parse(params.max_get_batch_size, json, key, false));
      |                                        ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/build/_deps/repo-backend-src/include/triton/backend/backend_common.h:97:38: note: in definition of macro 'RETURN_IF_ERROR'
   97 |     TRITONSERVER_Error* rie_err__ = (X); \
      |                                      ^
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:33,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:265:44: error: 'struct HugeCTR::PersistentDatabaseParams' has no member named 'max_get_batch_size'; did you mean 'max_batch_size'?
  265 |         "max. batch size (GET) = ", params.max_get_batch_size);
      |                                            ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/include/triton_common.hpp:39:45: note: in definition of macro 'HPS_TRITON_LOG'
   39 |     const std::string& msg = hps_str_concat(__VA_ARGS__);               \
      |                                             ^~~~~~~~~~~
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:28,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:269:40: error: 'struct HugeCTR::PersistentDatabaseParams' has no member named 'max_set_batch_size'; did you mean 'max_batch_size'?
  269 |         TritonJsonHelper::parse(params.max_set_batch_size, json, key, false));
      |                                        ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/build/_deps/repo-backend-src/include/triton/backend/backend_common.h:97:38: note: in definition of macro 'RETURN_IF_ERROR'
   97 |     TRITONSERVER_Error* rie_err__ = (X); \
      |                                      ^
In file included from /repos/hugectr_triton_backend/hps_backend/include/triton_helpers.hpp:33,
                 from /repos/hugectr_triton_backend/hps_backend/include/backend.hpp:38,
                 from /repos/hugectr_triton_backend/hps_backend/src/backend.cpp:29:
/repos/hugectr_triton_backend/hps_backend/src/backend.cpp:272:44: error: 'struct HugeCTR::PersistentDatabaseParams' has no member named 'max_set_batch_size'; did you mean 'max_batch_size'?
  272 |         "max. batch size (SET) = ", params.max_set_batch_size);
      |                                            ^~~~~~~~~~~~~~~~~~
/repos/hugectr_triton_backend/hps_backend/include/triton_common.hpp:39:45: note: in definition of macro 'HPS_TRITON_LOG'
   39 |     const std::string& msg = hps_str_concat(__VA_ARGS__);               \
      |                                             ^~~~~~~~~~~
make[2]: *** [CMakeFiles/triton-hps-backend.dir/build.make:90: CMakeFiles/triton-hps-backend.dir/src/backend.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:177: CMakeFiles/triton-hps-backend.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

Reply:
Hi @jperez999 , this error was due to hugectr_backend is not latest in github. Yingcan will push the changes to github today.

Reply:
> These files will need to be run against updated versions of triton and dlfw (23.01) and I am getting a failure in the following instruction `COPY --chown=1000:1000 --from=triton /usr/local/cuda-11.8/targets/x86_64-linux/lib/libcupti.so.11.8 /usr/local/cuda-11.8/targets/x86_64-linux/lib/libcupti.so.11.8`

I just aware I'm still building it with the version as merlin-base-prelease-pipe. After changed to use 23.01, yes, I met this failure...too.

Reply:
Removed other dependencies which are already in the image, so no need downgrade now.